File: backend/src/main/java/com/btalk/entity/Conversation.java
package com.btalk.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import com.btalk.constants.ConversationType;

import java.time.LocalDateTime;

@Entity
@Table(name = "conversations")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Conversation {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String conversationId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ConversationType type;

    private String name;

    @Column(name = "creator_id", nullable = false)
    private String creatorId;

    @CreationTimestamp
    private LocalDateTime createdAt;
}

File: frontend/src/app/models/conversation.ts
import { Message } from "./message";
import { Participant } from "./participant";

export interface Conversation {
  conversationId: string; // UUID as string
  type: 'PRIVATE' | 'GROUP';
  name?: string;
  creatorId: string; // UUID as string
  createdAt: string;
  participants: Participant[];
  lastMessage?: Message;
  unreadCount: number;
}
